torchtune/modules/kv_cache.py

```diff
@@ -17,50 +17,38 @@ class KVCache(nn.Module):
     Args:
         max_batch_size (int): maximum batch size model will be run with
         max_seq_len (int): maximum sequence length model will be run with
-        n_kv_heads (int): number of kv heads
+        num_heads (int): number of heads. We take num_heads instead of num_kv_heads because
+            the cache is created after we've expanded the key and value tensors to have the
+            same shape as the query tensor. See attention.py for more details
```

Comment:
So this makes sense to me, but given that you didn't really change what we were doing on the attention side of things, how was this working properly before?

Reply:
I think this is the case where we haven't really run inference for the case where num_kv_heads != num_heads. If we did, this would break